creator/ingest_runs/tasks.py

```diff
@@ -1,13 +1,109 @@
-def run_ingest(ingest_run_id):
-    # TODO
-    print(f"Preparing ingest run {ingest_run_id}")
+import django_rq
+import logging
 
+from django.core.exceptions import ValidationError
 
-def cancel_ingest(ingest_run_id):
-    # TODO
-    print(f"Cancelling ingest run {ingest_run_id}")
+from creator.decorators import task
+from creator.files.models import FileType
+from creator.ingest_runs.models import IngestRun
+from creator.utils import stop_job
+
+
+logger = logging.getLogger(__name__)
+
+
+@task("run_ingest")
+def run_ingest(ingest_run_id=None):
+    """
+    Run ingest process for a given set of files.
+    """
+    ingest_run = get_ingest_run(ingest_run_id)
+    logging.info(f"Preparing ingest run {ingest_run.pk} for processing.")
+
+    # Check if any IngestRuns with the same input hash exist. If so, cancel
+    # them and delete the newly created IngestRun, which never gets started.
+    # If not, start a new IngestRun.
+    if cancel_duplicate_ingest_runs(ingest_run):
+        ingest_run.delete()
+    else:
+        # Update run state so we know it's running
+        logging.info(f"Start ingest {ingest_run.id}")
+        ingest_run.start()
+        ingest_run.save()
+
+        # Check that the files are genomic workflow manifests
+        versions = ingest_run.versions.all()
+        are_gwo = (check_gwo(version) for version in versions)
+        if all(are_gwo):
+            logging.info(
+                "All files in the ingest run are genomic workflow manifests."
+            )
+            ingest_genomic_workflow_output_manifests(ingest_run)
```

Comment:
I think this line should probably be wrapped in a general try except so that we can catch anything that goes wrong fail the run, and then re-raise the exception to halt execution. This would handle anything unexpected that could happen during ingest, as well as job timeouts which cause an exception if the job duration takes longer than the default queue timeout.